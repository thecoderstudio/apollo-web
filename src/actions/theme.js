function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
