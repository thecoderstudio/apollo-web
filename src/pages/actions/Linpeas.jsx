import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import TerminalPage from '../TerminalPage';
import DownloadLinpeas from '../../components/action/DownloadLinpeas';
import Icon from '../../components/Icon';

const Container = styled.div`
  height: 100%;
`;

const Fab = styled.div`
  position: absolute;
  width: 55px;
  height: 55px;
  bottom: 30px;
  right: 30px;
  background-color: ${props => props.theme.primary};
  border-radius: 50%;
  z-index: 5;
  display: flex;
  align-items: center;
  cursor: pointer;

  * {
    margin: 0 auto;
  }
`;

class Linpeas extends React.PureComponent {
  constructor(props) {
    super(props);
    this.startExporting = this.startExporting.bind(this);
    this.stopExporting = this.stopExporting.bind(this);
    this.setComplete = this.setComplete.bind(this);
    this.state = {
      exporting: false,
      complete: false
    };
  }

  startExporting() {
    this.setState({
      exporting: true
    });
  }

  stopExporting() {
    this.setState({
      exporting: false
    });
  }

  setComplete(complete) {
    this.setState({
      complete
    });
  }

  render() {
    const { match: { params } } = this.props;
    const agent = this.props.agents.get(params.agentId);

    return (
      <Container>
        <TerminalPage 
          agentEndpoint='action/linpeas'
          onSocketClose={this.setComplete}
          readOnly {...this.props} />
        { this.state.complete && 
          <Fab onClick={this.startExporting}>
            <Icon className="fas fa-download" />
          </Fab>
        }
        { this.state.exporting && <DownloadLinpeas agent={agent} onClose={this.stopExporting} /> }
      </Container>
    );
  }
}

export default connect(
  state => ({agents: state.agent})
)(Linpeas);
