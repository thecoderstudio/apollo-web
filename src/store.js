import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
