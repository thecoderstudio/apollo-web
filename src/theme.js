export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
