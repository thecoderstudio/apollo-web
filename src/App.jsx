import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
