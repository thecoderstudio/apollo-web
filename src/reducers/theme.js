import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
