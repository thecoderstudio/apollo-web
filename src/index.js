import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
