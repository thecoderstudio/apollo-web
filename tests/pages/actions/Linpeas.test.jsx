import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import WS from 'jest-websocket-mock';
import { mount } from 'enzyme';
import Linpeas from '../../../src/pages/actions/Linpeas';
import { darkTheme } from '../../../src/theme';

const mockStore = configureStore([]);

function getComponentTags(store) {
  return (
    <div>
      <Provider store={store}>
        <ThemeProvider theme={darkTheme}>
          <Linpeas
            match={{
              params: {
                agentId: "fakeid"
              }
            }}
          />
        </ThemeProvider>
      </Provider>
    </div>
  );
}

describe('linpeas', () => {
  const server = new WS(`ws://localhost:1234/agent/fakeid/action/linpeas`);

  afterEach(() => {
    WS.clean();
  });

  it('correctly renders', async () => {
    const agents = new Map();
    agents.set('fakeid', {
      id: 'fakeid',
      name: 'fake',
      connectionState: 'connected'
    });
    const store = mockStore({
      agent: agents
    });
    const wrapper = mount(getComponentTags(store));
    await server.connected;
    expect(wrapper).toMatchSnapshot();
  });
});
