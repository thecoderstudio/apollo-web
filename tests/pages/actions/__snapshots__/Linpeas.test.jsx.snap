// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linpeas correctly renders 1`] = `
.c1 {
  height: 100%;
  width: 100%;
  padding: 20px;
}

.c2 {
  height: 100%;
  width: 100%;
}

.c2 .xterm-viewport {
  overflow: hidden !important;
}

.c0 {
  height: 100%;
}

<div>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "accent": "#4D02EE",
          "black": "#181818",
          "boxShadow": "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
          "connectedColor": "#27AE60",
          "connectingColor": "#F39C12",
          "darkGrey": "#3E3E3E",
          "darkWhite": "#F1F1F1",
          "disconnectedColor": "#C0392B",
          "error": "#ED3E2A",
          "green": "#32E67E",
          "inactive": "#363636",
          "lightBlack": "#1B1B1B",
          "name": "dark",
          "overlay": "rgba(37, 37, 37, 0.6)",
          "primary": "#02A3EE",
          "red": "#ED3E2A",
          "selectedBlack": "#252525",
          "smallTextSize": "0.85rem",
          "warning": "#ED852A",
          "white": "#F8F8F8",
        }
      }
    >
      <Connect(Linpeas)
        match={
          Object {
            "params": Object {
              "agentId": "fakeid",
            },
          }
        }
      >
        <Linpeas
          agents={
            Map {
              "fakeid" => Object {
                "connectionState": "connected",
                "id": "fakeid",
                "name": "fake",
              },
            }
          }
          dispatch={[Function]}
          match={
            Object {
              "params": Object {
                "agentId": "fakeid",
              },
            }
          }
        >
          <styled.div>
            <div
              className="c0"
            >
              <Connect(TerminalPage)
                agentEndpoint="action/linpeas"
                agents={
                  Map {
                    "fakeid" => Object {
                      "connectionState": "connected",
                      "id": "fakeid",
                      "name": "fake",
                    },
                  }
                }
                dispatch={[Function]}
                match={
                  Object {
                    "params": Object {
                      "agentId": "fakeid",
                    },
                  }
                }
                onSocketClose={[Function]}
                readOnly={true}
              >
                <TerminalPage
                  agentEndpoint="action/linpeas"
                  agents={
                    Map {
                      "fakeid" => Object {
                        "connectionState": "connected",
                        "id": "fakeid",
                        "name": "fake",
                      },
                    }
                  }
                  dispatch={[Function]}
                  match={
                    Object {
                      "params": Object {
                        "agentId": "fakeid",
                      },
                    }
                  }
                  onSocketClose={[Function]}
                  readOnly={true}
                >
                  <WithTheme(Terminal)
                    agent={
                      Object {
                        "connectionState": "connected",
                        "id": "fakeid",
                        "name": "fake",
                      }
                    }
                    agentEndpoint="action/linpeas"
                    onSocketClose={[Function]}
                    readOnly={true}
                  >
                    <Terminal
                      agent={
                        Object {
                          "connectionState": "connected",
                          "id": "fakeid",
                          "name": "fake",
                        }
                      }
                      agentEndpoint="action/linpeas"
                      onSocketClose={[Function]}
                      readOnly={true}
                      theme={
                        Object {
                          "accent": "#4D02EE",
                          "black": "#181818",
                          "boxShadow": "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
                          "connectedColor": "#27AE60",
                          "connectingColor": "#F39C12",
                          "darkGrey": "#3E3E3E",
                          "darkWhite": "#F1F1F1",
                          "disconnectedColor": "#C0392B",
                          "error": "#ED3E2A",
                          "green": "#32E67E",
                          "inactive": "#363636",
                          "lightBlack": "#1B1B1B",
                          "name": "dark",
                          "overlay": "rgba(37, 37, 37, 0.6)",
                          "primary": "#02A3EE",
                          "red": "#ED3E2A",
                          "selectedBlack": "#252525",
                          "smallTextSize": "0.85rem",
                          "warning": "#ED852A",
                          "white": "#F8F8F8",
                        }
                      }
                    >
                      <styled.div>
                        <div
                          className="c1"
                        >
                          <styled.div
                            data-testid="terminal"
                          >
                            <div
                              className="c2"
                              data-testid="terminal"
                            />
                          </styled.div>
                        </div>
                      </styled.div>
                    </Terminal>
                  </WithTheme(Terminal)>
                </TerminalPage>
              </Connect(TerminalPage)>
            </div>
          </styled.div>
        </Linpeas>
      </Connect(Linpeas)>
    </Component>
  </Provider>
</div>
`;
